### PR TITLE
Add ability to deploy with the OpenTelemetry collector instead of the Grafana Agent

### PR DESCRIPTION
**Pull request by petewall**
https://grafana.com/docs/grafana-cloud/monitor-infrastructure/kubernetes-monitoring/configuration/configure-infrastructure-manually/otel-collector/

### CODE REVIEW

File: .github/configs/ct.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ chart-dirs:
 chart-repos:
   - grafana=https://grafana.github.io/helm-charts
   - opencost=https://opencost.github.io/opencost-helm-chart
+  - opentelemetry=https://open-telemetry.github.io/opentelemetry-helm-charts
```
Comment (skl) [inline .github/configs/ct.yaml:9]: Should this new dep be added to your update cronjob action?

File: Makefile [hunk 2]
```diff
@@ -30,16 +35,29 @@ install-deps: scripts/install-deps.sh
 %/logs.river: %/output.yaml
 	yq -r "select(.metadata.name==\"k8smon-grafana-agent-logs\") | .data[\"config.river\"] | select( . != null )" $< > $@
 
+# OpenTelemetry Collector generated files
+#%/values-otel-collector.yaml: %/values.yaml
+#	yq --yaml-output '.["grafana-agent"].enabled=false | .["grafana-agent-logs"].enabled=false | .["opentelemetry-collector"].enabled=true | .["opentelemetry-collector-logs"].enabled=true' $< > $@
```
Comment (skl) [inline Makefile:40]: Still needed?

File: Makefile [hunk 3]
```diff
@@ -30,16 +35,29 @@ install-deps: scripts/install-deps.sh
...
 
 generate-agent-configs: $(METRIC_CONFIG_FILES) $(LOG_CONFIG_FILES)
 
-regenerate-agent-configs: clean-example-outputs generate-agent-configs
+generate-otel-configs: $(OTEL_METRIC_CONFIG_FILES) $(OTEL_LOG_CONFIG_FILES)
 
-clean-example-outputs:
-	rm -f $(OUTPUT_FILES)
+#clean-example-outputs:
+#	rm -f $(OUTPUT_FILES)
```
Comment (skl) [inline Makefile:59]: looks like not needed anymore

File: charts/k8s-monitoring/Chart.lock [hunk 4]
```diff
@@ -5,6 +5,12 @@ dependencies:
 - name: grafana-agent
   repository: https://grafana.github.io/helm-charts
   version: 0.19.0
+- name: opentelemetry-collector
+  repository: https://open-telemetry.github.io/opentelemetry-helm-charts
+  version: 0.66.4
+- name: opentelemetry-collector
+  repository: https://open-telemetry.github.io/opentelemetry-helm-charts
+  version: 0.66.4
```
Comment (skl) [inline charts/k8s-monitoring/Chart.lock:13]: Duplicate?

File: charts/k8s-monitoring/charts/prometheus-node-exporter/.helmignore
Comment (skl) [inline charts/k8s-monitoring/charts/prometheus-node-exporter/.helmignore:1]: Looks like maybe the extracted node exporter chart got added, is that intentional?

File: examples/extra-rules/values-otel-collector.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  prometheus:
+    host: https://prometheus.example.com
+    basicAuth:
+      username: 12345
+      password: It's a secret to everyone
+    externalLabels:
+      site: northwest
+#    writeRelabelConfigRules: "write_relabel_config {\n  source_labels = [\"__name__\"\
+#      ]\n  regex = \"metric_to_drop|another_metric_to_drop\"\n  action = \"drop\"\n\
+#      }"
```
Comment (skl) [inline examples/extra-rules/values-otel-collector.yaml:13]: Is this WIP/TODO?

**Comment by petewall** [comment]
This is getting a major change. Closing the PR for now.

**Pull request closed without merging**